File: sql/plsql/CSharp/PlSqlParserBase.cs
using System;
using System.IO;
using System.Reflection;
using Antlr4.Runtime;
using Antlr4.Runtime.Misc;

public abstract class PlSqlParserBase : Parser
{
    private bool _isVersion10 = false;
    private bool _isVersion11 = true;
    private bool _isVersion12 = true;

    protected PlSqlParserBase(ITokenStream input)
        : base(input)
    {
    }

    public PlSqlParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput) : this(input)
    {
    }

    public bool isVersion10() => _isVersion10;

    public bool isVersion11() => _isVersion11;

    public bool isVersion12() => _isVersion12;

    public bool setVersion10(bool value) => _isVersion10 = value;

    public bool setVersion11(bool value) => _isVersion11 = value;

    public bool setVersion12(bool value) => _isVersion12 = value;

    public bool IsNotNumericFunction() {
	    var lt1 = (this.InputStream as CommonTokenStream).LT(1);
	    var lt2 = (this.InputStream as CommonTokenStream).LT(2);
	    if ( (lt1.Type == PlSqlParser.SUM ||
		lt1.Type == PlSqlParser.COUNT ||
		lt1.Type == PlSqlParser.AVG ||
		lt1.Type == PlSqlParser.MIN ||
		lt1.Type == PlSqlParser.MAX ||
		lt1.Type == PlSqlParser.ROUND ||
		lt1.Type == PlSqlParser.LEAST ||
		  lt1.Type == PlSqlParser.GREATEST) && lt2.Type == PlSqlParser.LEFT_PAREN)
		    return false;
	    return true;
    }
}

File: sql/plsql/Antlr4ng/PlSqlParserBase.ts
import { Parser, TokenStream, CommonTokenStream, Recognizer } from 'antlr4ng';
import { PlSqlLexer } from './PlSqlLexer.js';

export abstract class PlSqlParserBase extends Parser {

  _isVersion10: boolean;
  _isVersion11: boolean;
  _isVersion12: boolean;

  constructor(input: TokenStream) {
    super(input);
    this._isVersion10 = true;
    this._isVersion11 = true;
    this._isVersion12 = true;
  }

  isVersion10(): boolean {
    return this._isVersion10;
  }

  isVersion11(): boolean {
    return this._isVersion11;
  }

  isVersion12(): boolean {
    return this._isVersion12;
  }

  setVersion10(value: boolean): void {
    this._isVersion10 = value;
  }

  setVersion11(value: boolean): void {
    this._isVersion11 = value;
  }

  setVersion12(value: boolean): void {
    this._isVersion12 = value;
  }

  IsNotNumericFunction(): boolean {
    const lt1 = (this.tokenStream as CommonTokenStream).LT(1);
    const lt2 = (this.tokenStream as CommonTokenStream).LT(2);
    if ((lt1!.type === PlSqlLexer.SUM ||
      lt1!.type === PlSqlLexer.COUNT ||
      lt1!.type === PlSqlLexer.AVG ||
      lt1!.type === PlSqlLexer.MIN ||
      lt1!.type === PlSqlLexer.MAX ||
      lt1!.type === PlSqlLexer.ROUND ||
      lt1!.type === PlSqlLexer.LEAST ||
      lt1!.type === PlSqlLexer.GREATEST) && lt2!.type === PlSqlLexer.LEFT_PAREN)
      return false;
    return true;
  }

}